README.md

```diff
@@ -62,6 +62,29 @@ To set up a route, you can do the following:
 ```
 The plugin will render the component server side and return it, where as the route handler will return the props to the frontend when needed.
 
+## Imperatively rendering a component
+Imperative rendering allows routes to dynamically render a component based on specific conditions, instead of always rendering the same component. To do so, we still require that the component is registered to allow Vite to bundle it.
+
+Note that the route which renders the component is a normal route that doesn't need any special route options configuration.
+
+To register a component, you can do the following:
+```js
+  server.registerRenderable(require.resolve('./ImperativelyRenderablePage'))
+```
+
+And then you can render it imperatively in your routes:
+```js
+  server.get('/imperative/:bool', async (request: FastifyRequest<{ Params: { bool: string } }>, reply) => {
+    if (request.params.bool == 'true') {
+      return reply.render(require.resolve('./ImperativelyRenderablePage'), {
+        hostname: os.hostname(),
+        requestIP: request.ip,
+      })
+    } else {
+      return reply.redirect('/not-found')
+    }
+  })
+```
```

Comment:
Could you also document the shortcomings of imperative rendering? Once we know what they are I guess.

Reply:
Will create an issue for this.